### PR TITLE
fix failing ci

### PR DESCRIPTION
**Pull request by jhamman**
some minor cleanup and a small fix to a failing test (I'll highlight below).

### CODE REVIEW

File: xbatcher/generators.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def _drop_input_dims(ds, input_dims, suffix='_input'):
         out = out.rename({dim: newdim})
         # extra steps needed if there is a coordinate
         if newdim in out:
-            out = out.drop(newdim)
+            out = out.drop_vars(newdim)
```
Comment (jhamman) [inline xbatcher/generators.py:52]: the switch to `drop_vars` is in response to a pending depreciation of `drop` in xarray.

File: xbatcher/tests/test_generators.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def test_batch_3d_1d_input(sample_ds_3d, bsize):
         expected_slice = slice(bsize * n, bsize * (n + 1))
         ds_batch_expected = (
             sample_ds_3d.isel(x=expected_slice)
-            .stack(sample=['y', 'time'])
+            .stack(sample=['time', 'y'])
```
Comment (jhamman) [inline xbatcher/tests/test_generators.py:121]: this is the only substantive change in this PR.

Comment (jhamman) [reply xbatcher/tests/test_generators.py:121]: I take that back. See the change in 8df83c1 which didn't make the initial cut.

File: xbatcher/generators.py [hunk 3]
```diff
@@ -49,13 +49,13 @@ def _drop_input_dims(ds, input_dims, suffix='_input'):
...
             out.coords[dim] = newdim, ds[dim].data, ds[dim].attrs
     return out
 
 
 def _maybe_stack_batch_dims(ds, input_dims, stacked_dim_name='sample'):
-    batch_dims = list(set(ds.dims) - set(input_dims))
+    batch_dims = [d for d in ds.dims if d not in input_dims]
```
Comment (jhamman) [inline xbatcher/generators.py:58]: @rabernat / @tcchiao - this was a tricky one. The set difference, as was previously implemented, was not resolving deterministically. Sometimes, we would get `[‘time’, ‘y’]`, and others we would get `[‘y’, ‘time’]`. This new method is probably about as fast and is easier to read (in my opinion).

**Review by tcchiao — APPROVED** [review]
looks good to me, thanks for fixing this!

**Pull request merged**